rules/legacy/src/Rector/Include_/AddTopIncludeRector.php

```diff
@@ -181,20 +154,48 @@ private function initialize(): void
         $reflection = new ReflectionClass(Include_::class);
         $constants = $reflection->getConstants();
 
-        if (! isset($constants[$this->configuration['type'] ?? ''])) {
-            throw new InvalidRectorConfigurationException('Invalid type: must be one of ' . implode(
+        if (! isset($constants[$this->settings['type'] ?? ''])) {
+            throw new InvalidRectorsettingsException('Invalid type: must be one of ' . implode(
                 ', ',
                 array_keys($constants)
             ));
         }
 
-        if ('' === ($this->configuration['file'] ?? '')) {
-            throw new InvalidRectorConfigurationException('Invalid parameter: file must be provided');
+        if ('' === ($this->settings['file'] ?? '')) {
+            throw new InvalidRectorsettingsException('Invalid parameter: file must be provided');
+        }
+        $this->type = $constants[$this->settings['type']];
+        $this->file = $this->settings['file'];
+        $this->fileExpression = $this->getExpressionFromString($this->file);
+        // normalize the file we are including
+        $this->file = $this->getStringFromExpression($this->fileExpression);
+        if (isset($this->settings['match']) && is_array($this->settings['match'])) {
+            $this->patterns = $this->settings['match'];
+        }
+    }
+
+    private function getExpressionFromString(string $source): ?Expr
+    {
+        $ast = $this->parser->parse("<?php {$source};\n");
```

Comment:
Better use `$nodes` convention, so we have the same naming in all files

Reply:
Also, the "in string {$source}" should be prevented too.

```suggestion
        $nodes = $this->parser->parse('<?php ' . $source . ';');
```

 It makes PHPStan + Rector refactoring more complicated.


PHPStan missed that?